### PR TITLE
Updating the REPL tutorial

### LINKED ISSUE #339: Update the REPL documentation
https://github.com/informalsystems/tnt/blob/main/doc/repl.md is seriously outdated

### PR DESCRIPTION
**Pull request by konnov**
Closes #339. This is an update of the REPL tutorial:

 - [x] It is now written in the literate programming style (kind of) with `lmt`
 - [x] All definitions are extracted in the module `kettle.qnt`
 - [x] Most of the REPL interactions are extracted in the file `replTest.txt`, which is postprocessed in `Makefile`
   - The postprocessing is a bit adhoc. It would be great to make this approach more systematic in the future. In principle, it is not so far from notebooks. 
 - [x] An integration test replays the session from `replTestIn.txt` with `quint` and compares the output to `replTestOut.txt`.
 - [x] Quint now supports the flag `-q`, which suppresses the noisy prompts `>>> ` and `... `. Great for testing.
 - [x] REPL does not output line numbers anymore. There is no good understanding of which line numbers should be used: relative to the last expression, relative to the invisible buffer, relative to the input file, etc. It's also hard to do reliable scripting, when line numbers are printed in REPL.
 - [x] Tests added for any new code

### CODE REVIEW

**Comment by konnov** [comment]
For some reason, diff is failing on windows. Perhaps, I call it wrong way.

**Comment by konnov** [comment]
@shonfeder, perhaps you know how to fix this problem with Windows, as you had it with `txm` once.

**Comment by shonfeder** [comment]
I recommend just not running the diff check on windows. That’s how I’ve dealt with the other tests that requiring doffing.

File: quint/cli-tests.md [hunk 1]
```diff
@@ -211,3 +211,16 @@ Temporarily disabled.
...
+The REPL tutorial is reproducible in REPL.
+
+<!-- !test check REPL tutorial -->
+```
+if [ "$UNAME" == "Linux" -o "$UNAME" == "Darwin" ]; then
+  quint -q -r ../tutorials/repl/kettle.qnt::kettle <../tutorials/repl/replTestIn.txt \
+    | diff - ../tutorials/repl/replTestOut.txt
+fi
+# else diff does not work as expected on windows
+```
```
Comment (bugarela) [inline quint/cli-tests.md:226]: I'd rather move this to `io-cli-tests.md` which are not run on windows anyway. Then we can get rid of the `if` statement.

**Review by bugarela — APPROVED** [review]
This is a nice improvement on structure! I have only a suggestion on testing that I think we should implement.

File: quint/cli-tests.md (hunk 1 above)
Comment (konnov) [reply quint/cli-tests.md:226]: oh, I knew that one of them should be win-free :)

**Pull request merged**